Dev guide contextualization | Content release #2 | Go SDK (#2334)
temporalio/documentation: docs-src/go/choose-dev-cluster.md

---
id: choose-dev-cluster

> title: Choose a development Cluster

Reviewer: Cloud and self hosted are recommended in this section for production, but this section is about development. We could greatly simplify the whole Project setup section by moving Cloud/self-hosted options to a production section. (since it also shows up under "how to run a cloud worker" and CLI sections)

---
id: choose-dev-cluster
title: Choose a development Cluster
sidebar_label: Choose dev Cluster
description: Discover which development Cluster you should choose
tags:
  - developer guide
  - go sdk
---

**Which development Cluster should you choose?**

We recommend choosing a development environment based on your requirements.

> The source code for the Temporal Server (the orchestrating component of the Temporal Cluster) has an MIT open source license. So, in theory, anyone can take the Temporal Server code and run their Temporal Platform in any number of creative ways.

Reviewer 2: ```suggestion
The source code for the Temporal Server (the orchestrating component of the Temporal Cluster) is licensed under the MIT open source license.. So, in theory, anyone can take the Temporal Server code and run their Temporal Platform in any number of creative ways.
```
Activie voice

The Temporal CLI comes bundled with a development server and provides a fast way to start running Temporal Applications.

However, the local development server does not emit any metrics.
If you are eager to to set up Cluster-level metrics for performance tuning, we recommend using a self-hosted Cluster or Temporal Cloud.

#### Start the dev server

**How to start a local development server**

If you have successfully installed the Temporal CLI, open a new terminal and run the following command:

```bash
temporal server start-dev
```

This command automatically starts the Temporal Web UI, creates a default Namespace, and creates an in-memory database.

The Temporal Web UI serves to [http://localhost:8233](http://localhost:8233/).

For more command details and options, see the [CLI reference](/cli/server/start-dev)

#### Create a custom Namespace

**How to create a new Namespace on the development server**

The development server does automatically create a default Namespace (named "default") when it starts up.

> However, we will create a custom one for our application.

Reviewer 2: ```suggestion
However, you will create a custom one for our application.
```

https://developers.google.com/style/person

> Since this is something we recommend at a production level, we recommend practicing it with the development server.

Reviewer 2: ```suggestion
Since this is something recommended at a production level, it's recommend practicing it with the development server.
```

Use the `temporal operator namespace create` command using the Temporal CLI to create a new Namespace on the development server.

```bash
temporal operator namespace create backgroundcheck_namespace
```

For command details and options, see the [CLI reference](/cli/operator/namespace/create).

### Temporal Cloud

**When to use Temporal Cloud**

> If you do not have a Temporal Cloud Account, you can request one using the link on the [get started](https://docs.temporal.io/cloud/get-started) page.

Author: ```suggestion
If you do not have a Temporal Cloud Account, you can request one using the link on the [Get started with Temporal Cloud](https://docs.temporal.io/cloud/get-started) guide.
```

We recommend starting off with Temporal Cloud if you already have a production use case, or need to move a scalable proof of concept into production.

In other words, Temporal Cloud is perfect if you are ready to run at scale and don’t want the overhead of managing your own self-hosted Cluster.

> To create a Namespace in Temporal Cloud, follow the instructions in [How to create a Namespace](/cloud/namespaces-create).

Reviewer 2: `get started` is lower case but `How to create a Namespace` is upper case.
They're both titles, should they both be capitalized?

:::info Safely store your certificate and private key

> Store your certificate and private key you generated for your Namespace as files or environment variables in your project.

Reviewer 2: ```suggestion
Store certificates and private keys generated for your Namespace as files or environment variables in your project.
```

> You will need access to your certificate and key to run your Workers and start Workflows.

Reviewer 2: ```suggestion
You need access to your certificate and key to run your Workers and start Workflows.
```

> For certificate requirements and details, check out [How to manage certificates in Temporal Cloud](/cloud/certificates-intro).

Reviewer 2: ```suggestion
For more information on certificate requirements, see [How to manage certificates in Temporal Cloud](/cloud/certificates-intro).
```

we should just standardize the way we link out.

For example, running a self-hosted Cluster lets you try different databases, view Cluster metrics, use custom [Search Attributes](/concepts/what-is-a-search-attribute), and even play with the [Archival](/concepts/what-is-archival) feature.

It's completely possible to run a production level self-hosted Cluster.
However, for the purposes of this guide, we show how to use a self-hosted environment that runs completely out of Docker.
We acknowledge that it takes a fair amount of experience to elevate from a self-hosted environment in Docker to something that can run at an enterprise production scale.
The self-hosted information in this guide should help you make more informed decisions.

To follow along with self-hosted parts of this guide, install the following:

- [Docker](https://docs.docker.com/engine/install)
- [Docker Compose](https://docs.docker.com/compose/install).

Then, clone the [temporalio/docker-compose](https://github.com/temporalio/docker-compose.git) repository.

Change directory into the root of the project.

> Run the `docker-compose up` command.

Reviewer 2: ```suggestion
Run the `docker compose up` command.
```

newer docker

```shell
git clone https://github.com/temporalio/docker-compose.git
cd  docker-compose

> docker-compose up

Reviewer 2: ```suggestion
docker compose up
```
new docker

```

Create a command alias for the Temporal CLI:

```shell
alias temporal_docker="docker exec temporal-admin-tools temporal"
```

> Create a new Namespace.

Reviewer 2: ```suggestion
Create a Namespace.
```
